numba/dispatcher.py

```diff
@@ -433,23 +438,49 @@ def __reduce__(self):
             sigs = [cr.signature for cr in self.overloads.values()]
         globs = self._compiler.get_globals_for_reduction()
         return (serialize._rebuild_reduction,
-                (self.__class__, serialize._reduce_function(self.py_func, globs),
+                (self.__class__, str(self._uuid),
+                 serialize._reduce_function(self.py_func, globs),
                  self.locals, self.targetoptions, self._impl_kind,
                  self._can_compile, sigs))
 
     @classmethod
-    def _rebuild(cls, func_reduced, locals, targetoptions, impl_kind,
+    def _rebuild(cls, uuid, func_reduced, locals, targetoptions, impl_kind,
                  can_compile, sigs):
         """
         Rebuild an Dispatcher instance after it was __reduce__'d.
         """
+        try:
+            return cls._memo[uuid]
+        except KeyError:
+            pass
         py_func = serialize._rebuild_function(*func_reduced)
         self = cls(py_func, locals, targetoptions, impl_kind)
+        # Make sure this deserialization will be merged with subsequent ones
+        self._set_uuid(uuid)
         for sig in sigs:
             self.compile(sig)
         self._can_compile = can_compile
         return self
 
+    @property
+    def _uuid(self):
+        """
+        An instance-specific UUID, to avoid multiple deserializations of
+        a given instance.
+
+        Note this is lazily-generated, for performance reasons.
```

Comment:
Not really on most mainstream systems, I guess. The `uuid` module can do some weird stuff on some systems, though (such as launch a system executable to get the MAC address).